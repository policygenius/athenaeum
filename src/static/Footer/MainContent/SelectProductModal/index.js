import React from 'react';
import PropTypes from 'prop-types';

import Modal from 'molecules/Modal';
import Button from 'atoms/Button';
import Text from 'atoms/Text';
import SelectField from 'molecules/formfields/SelectField';
import Spacer from 'atoms/Spacer';

const selection = [
  {
    label: 'Life Insurance',
    value: '/life-insurance'
  },
  {
    label: 'Health Insurance',
    value: '/health-insurance'
  },
  {
    label: 'Long-term Disability Insurance',
    value: '/long-term-disability-insurance'
  },
  {
    label: 'Pet Insurance',
    value: '/pet-insurance'
  },
  {
    label: 'Renters Insurance',
    value: '/renters-insurance'
  },
  {
    label: 'Auto Insurance',
    value: '/auto-insurance'
  },
];

class SelectProductModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      product: 'life-insurance'
    };
  }

  handleSelect = (event) => {
    this.setState({
      product: event.target.value
    });
  }

  render() {
    const {
      modalProps
    } = this.props;

    return (
      <Modal
        contentLabel='Modal'
        {...modalProps}
      >
        <div>
          <Text size={6}>What insurance are you shopping for?</Text>
          <Spacer small />
          <SelectField
            placeholder='Select a Product'
            selectOptions={selection}
            input={{
              onChange: this.handleSelect
            }}
          />
          <Spacer small />
          <Button href={this.state.product}>
            Get Quotes
          </Button>
        </div>
      </Modal>
    );
  }
}

SelectProductModal.propTypes = {
  modalProps: PropTypes.object,
};

SelectProductModal.defaultProps = {
  // Place any default props here.
};

export default SelectProductModal;
