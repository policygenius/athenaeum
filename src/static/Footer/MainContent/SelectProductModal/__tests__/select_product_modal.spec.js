import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SelectProductModal from '../';

describe('<SelectProductModal />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<SelectProductModal />);

    expect(wrapper.type()).toEqual(SelectProductModal);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<SelectProductModal />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
