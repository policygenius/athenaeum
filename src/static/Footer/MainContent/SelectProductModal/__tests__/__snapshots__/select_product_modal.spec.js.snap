// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectProductModal /> renders correctly 1`] = `<noscript />`;
